### PR TITLE
[DI] use try-finally for container

### PR DESCRIPTION
**Pull request by Tobion**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

Continuation of #15949

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Container.php [hunk 1]
```diff
@@ -272,17 +272,13 @@ public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE
             try {
                 $service = $this->$method();
             } catch (\Exception $e) {
-                unset($this->loading[$id]);
-
-                if (array_key_exists($id, $this->services)) {
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Container.php]: if this call is not necessary, we could remove it in older branches too

Comment (Tobion) [inline src/Symfony/Component/DependencyInjection/Container.php]: we could yes

Comment (Tobion) [inline src/Symfony/Component/DependencyInjection/Container.php]: #16691

**Comment by fabpot** [comment]
Thank you @Tobion.

**Pull request merged**